cli/cmd/generate_azure.go

```diff
@@ -0,0 +1,680 @@
+		&GenerateAzureCommandState.Config, "config", false, "enable config integration")
+
+	generateAzureTfCommand.PersistentFlags().StringVar(
+		&GenerateAzureCommandState.ConfigIntegrationName,
+		"config_name",
+		"",
+		"specify name of config, if none given will default to TF config")
+
+	generateAzureTfCommand.PersistentFlags().BoolVar(
+		&GenerateAzureCommandState.CreateAdIntegration, "ad_create", true, "create new active directory integration")
+	generateAzureTfCommand.PersistentFlags().BoolVar(
+		&GenerateAzureCommandState.ManagementGroup, "mgmt_group", false, "use management rather than subscriptions")
+	generateAzureTfCommand.PersistentFlags().StringVar(
+		&GenerateAzureCommandState.ManagementGroupId,
+		"mgmt_group_id",
+		"",
+		"specify managment group id")
+
+	generateAzureTfCommand.PersistentFlags().BoolVar(
+		&GenerateAzureCommandState.ExistingStorageAccount, "existing_storage", false, "use existing storage account")
+	generateAzureTfCommand.PersistentFlags().StringVar(
+		&GenerateAzureCommandState.StorageAccountName,
+		"strg_account_name",
+		"",
+		"specify storage account name")
+	generateAzureTfCommand.PersistentFlags().StringVar(
+		&GenerateAzureCommandState.StorageAccountResourceGroup,
+		"strg_res_group",
+		"",
+		"specify storage resource group")
+	generateAzureTfCommand.PersistentFlags().StringVar(
+		&GenerateAzureCommandState.StorageLocation,
+		"location",
+		"",
+		"specify azure region where storage account logging resides")
+
+	generateAzureTfCommand.PersistentFlags().BoolVar(
+		&GenerateAzureCommandState.AllSubscriptions, "all_subs", false, "grant read access to ALL subscriptions within Tenant")
+
+	generateAzureTfCommand.PersistentFlags().StringSliceVar(
+		&GenerateAzureCommandState.SubscriptionIds,
+		"sub_ids",
+		[]string{},
+		`list of subscriptions to grant read access; format is id1,id2,id3`)
+
+	generateAzureTfCommand.PersistentFlags().StringVar(
+		&GenerateAzureCommandState.AdApplicationPassword,
+		"ad_pass",
+		"",
+		"password for an already existing active directory application")
+
+	generateAzureTfCommand.PersistentFlags().StringVar(
+		&GenerateAzureCommandState.AdApplicationId,
+		"ad_id",
+		"",
+		"id for an already existing active directory application")
+
+	generateAzureTfCommand.PersistentFlags().StringVar(
+		&GenerateAzureCommandState.AdServicePrincipalId,
+		"ad_pid",
+		"",
+		"principle id for an already existing active directory application")
+
+	generateAzureTfCommand.PersistentFlags().BoolVar(
+		&GenerateAzureCommandExtraState.TerraformApply, "terraform-apply", false, "run terraform apply for the generated hcl")
+
+}
+
+func promptAzureIntegrationNameQuestions(config *azure.GenerateAzureTfConfigurationArgs) error {
+
+	if err := SurveyMultipleQuestionWithValidation([]SurveyQuestionWithValidationArgs{
+		{
+			Prompt:   &survey.Input{Message: QuestionAzureConfigName, Default: config.ConfigIntegrationName},
+			Checks:   []*bool{&config.Config},
+			Required: true,
```

Comment:
We should probably remove `Required` for this question. What if the user only wants to supply a custom name for their Activity log integration and not their config integration?

Reply:
Done